packages/vscode-extension/src/handlers.ts

```diff
@@ -908,6 +910,38 @@ async function openMarkdownHandler() {
   }
 }
 
+async function postUpgrade(): Promise<void> {
+  await openUpgradeChangeLogsHandler();
+  await popupAfterUpgrade();
+}
+
+async function popupAfterUpgrade(): Promise<void> {
+  const aadClientSecretFlag = "NeedToSetAADClientSecretEnv";
+  const aadClientSecret = globalStateGet(aadClientSecretFlag, "");
+  if (
+    aadClientSecret !== "" &&
+    workspace.workspaceFolders &&
+    workspace.workspaceFolders.length > 0
+  ) {
+    try {
+      const learnMoreLink = StringResources.vsc.upgradeToMultiEnvAndBicep.learnMoreLink;
+      const learnMoreText = StringResources.vsc.upgradeToMultiEnvAndBicep.learnMoreText;
+      const option = { modal: false };
+
+      const outputMsg = `Please set "AAD_APP_CLIENT_SECRET" = "${aadClientSecret}" in your system environment variable. Read this wiki(${learnMoreLink}) to for more details.`;
+      VsCodeLogInstance.warning(outputMsg);
+      const showMsg = `Please set "AAD_APP_CLIENT_SECRET" = "${aadClientSecret}" in your system environment variable.\nYou can also click Learn More for more details.`;
```

Comment:
Same here. The principle is whether the string is visible to users, if the answer is yes, then you should extract to Strings.json file.

Reply:
ok

Reply:
fixed